### PR TITLE
Add local `SubprocessCluster` that runs workers in separate processes

### PR DESCRIPTION
**Pull request by hendrikmakait**
Closes #xxxx

- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by hendrikmakait** [comment]
This is currently broken on Windows because we use the `SelectorEventLoop` on Windows, which does not support subprocesses (https://docs.python.org/3/library/asyncio-platforms.html#windows). Using `SelectorEventLoop` seems to be necessary due to an old issue with `tornado` (https://github.com/dask/distributed/blob/d87cea954aed8d2e9a49fc88fe0bf4aace805f08/distributed/config.py#L185-L189). 

Upgrading to `tornado>=6.1` might resolve this, is there a reason #5833 has not been merged? cc @graingert

File: distributed/deploy/subprocess.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        )
+    assert n_workers
+
+    scheduler_options.update(
+        {
+            "host": host,
+            "port": scheduler_port,
+            "dashboard": dashboard_address is not None,
+            "dashboard_address": dashboard_address,
+        }
```
Comment (mrocklin) [inline distributed/deploy/subprocess.py:100]: These keys might already be in `scheduler/worker_options`.  Maybe we want to use the `dict.setdefault` method instead

**Review by mrocklin — APPROVED** [review]
In principle this looks sensible to me.  I would add basic docstrings.

Maybe also a single line mentioning this class at the end of https://docs.dask.org/en/stable/deploying-python.html#localcluster ?

**Comment by mrocklin** [comment]
> This is currently broken on Windows because we use the SelectorEventLoop on Windows, which does not support subprocesses (https://docs.python.org/3/library/asyncio-platforms.html#windows). Using SelectorEventLoop seems to be necessary due to an old issue with tornado (

For something like this that's already supporting an odd corner case, I think that it's totally fine to just raise an informative error message on Windows and move on.

**Comment by hendrikmakait** [comment]
> For something like this that's already supporting an odd corner case, I think that it's totally fine to just raise an informative error message on Windows and move on.

Added a descriptive error message 👌

**Comment by graingert** [comment]
> Upgrading to `tornado>=6.1` might resolve this, is there a reason #5833 has not been merged? cc @graingert

I was waiting for a bokeh release so it might be worth resurrecting that PR

**Comment by mrocklin** [comment]
Thanks @hendrikmakait .  Seems good to me. I'm going to wait until end of day to see if we get any user feedback and then I'll merge in.

**Comment by tasansal** [comment]
This is great; one benefit I can think of is the overhead of going from the scheduler process back to the client process disappears. What are other benefits / use cases? I wouldn't mind testing this further and adding some more to the documentation, which is very little now.

**Comment by mrocklin** [comment]
In some situations creating workers with the multiprocessing library can
cause issues.  This is relatively rare though and not something we want to
focus on that much.

**Pull request merged**